sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -631,8 +631,7 @@ object ScalaReflection extends ScalaReflection {
               "cannot be used as field name\n" + walkedTypePath.mkString("\n"))
           }
 
-          val fieldValue = Invoke(
-            AssertNotNull(inputObject, walkedTypePath), fieldName, dataTypeFor(fieldType),
+          val fieldValue = Invoke(inputObject, fieldName, dataTypeFor(fieldType),
```

Comment:
let's add a comment to say that, the `inputObject` won't be null because of the `If`